src/main/scala/sigmastate/serialization/SerializedConstantPlaceholderStore.scala

```diff
@@ -0,0 +1,57 @@
+package sigmastate.serialization
+
+import sigmastate.SType
+import sigmastate.Values.Constant
+import sigmastate.lang.SigmaBuilder
+import sigmastate.utils.{ByteReader, ByteWriter}
+
+import scala.collection.mutable
+
+class SerializedConstantPlaceholderStore(builder: SigmaBuilder) {
+
+  private val store = mutable.Map[Constant[SType], Int]()
+  private val reversedStore = mutable.Map[Int, Constant[SType]]()
```

Comment:
@greenhat we just need an array here

Reply:
@ergomorphic Nice catch! Thanks. Good ol' array, how could I forget about you. :)